src/sagemaker/jumpstart/artifacts.py

```diff
@@ -0,0 +1,257 @@
+        region, model_id, model_version
+    )
+
+    if image_scope == INFERENCE:
+        ecr_specs = model_specs.hosting_ecr_specs
+    elif image_scope == TRAINING:
+        if not model_specs.training_supported:
+            raise ValueError(f"JumpStart model id '{model_id}' does not support training.")
+        assert model_specs.training_ecr_specs is not None
+        ecr_specs = model_specs.training_ecr_specs
+
+    if framework is not None and framework != ecr_specs.framework:
+        raise ValueError(f"Bad value for container framework for JumpStart model: '{framework}'.")
+
+    if version is not None and version != ecr_specs.framework_version:
+        raise ValueError(
+            f"Bad value for container framework version for JumpStart model: '{version}'."
+        )
+
+    if py_version is not None and py_version != ecr_specs.py_version:
+        raise ValueError(
+            f"Bad value for container python version for JumpStart model: '{py_version}'."
+        )
+
+    base_framework_version_override = None
+    version_override = None
+    if ecr_specs.framework == ModelFramework.HUGGINGFACE.value:
+        base_framework_version_override = ecr_specs.framework_version
+        version_override = ecr_specs.huggingface_transformers_version
+
+    if image_scope == TRAINING:
+        return image_uris.get_training_image_uri(
+            region=region,
+            framework=ecr_specs.framework,
+            framework_version=version_override or ecr_specs.framework_version,
+            py_version=ecr_specs.py_version,
+            image_uri=None,
+            distribution=None,
+            compiler_config=None,
+            tensorflow_version=None,
+            pytorch_version=base_framework_version_override or base_framework_version,
+            instance_type=instance_type,
+        )
+    if base_framework_version_override is not None:
+        base_framework_version_override = f"pytorch{base_framework_version_override}"
+
+    return image_uris.retrieve(
+        framework=ecr_specs.framework,
+        region=region,
+        version=version_override or ecr_specs.framework_version,
+        py_version=ecr_specs.py_version,
+        instance_type=instance_type,
+        accelerator_type=accelerator_type,
+        image_scope=image_scope,
+        container_version=container_version,
+        distribution=distribution,
+        base_framework_version=base_framework_version_override or base_framework_version,
+        training_compiler_config=training_compiler_config,
+    )
+
+
+def _retrieve_model_uri(
+    model_id: str,
+    model_version: str,
+    model_scope: Optional[str],
+    region: Optional[str],
+):
+    """Retrieves the model artifact S3 URI for the model matching the given arguments.
+
+    Args:
+        model_id (str): JumpStart model id for which to retrieve model S3 URI.
+        model_version (str): JumpStart model version for which to retrieve model S3 URI.
```

Comment:
"Version of the JumpStart model for which to retrieve the model artifact S3 URI."